File: smoketests/tests/energy.py
from .. import Smoketest, requires_anonymous_login
import time

class EnergyFlow(Smoketest):

    @requires_anonymous_login
    def test_energy_balance(self):
        """Test getting energy balance."""

        self.new_identity()
        self.publish_module()

        out = self.spacetime("energy", "balance")
        self.assertRegex(out, '{"balance":"-?[0-9]+"}')

File: crates/smoketests/tests/energy.rs
use regex::Regex;
use spacetimedb_smoketests::Smoketest;

/// Test getting energy balance.
#[test]
fn test_energy_balance() {
    let test = Smoketest::builder().build();

    let output = test
        .spacetime(&["energy", "balance", "--server", &test.server_url])
        .unwrap();
    let re = Regex::new(r#"\{"balance":"-?[0-9]+"\}"#).unwrap();
    assert!(re.is_match(&output), "Expected energy balance JSON, got: {}", output);
}
